package-res/resources/web/vizapi/VizController.js

```diff
@@ -716,29 +716,31 @@ pentaho.VizController.getRgbGradientFromMultiColorHex = function(value, min, max
   var steps = colors.length-1;
   var range = max-min;
 
+  var startStep;
+  var endStep;
   if(range <= 0) {
-    var start = colors.length-1;
-    var end = start;
+    startStep = colors.length-1;
+    endStep = startStep;
   } else {
-    var start = Math.floor(((value-min)/range) * steps);
-    var end = Math.ceil(((value-min)/range) * steps);
+    var inRange = ((value-min)/range);
+    startStep = Math.floor(inRange * steps);
+    endStep = Math.ceil(inRange * steps);
   }
-  var color1 = pentaho.VizController.convertToRGB(colors[start]);
-  var color2 = pentaho.VizController.convertToRGB(colors[end]);
+  var color1 = pentaho.VizController.convertToRGB(colors[startStep]);
+  var color2 = pentaho.VizController.convertToRGB(colors[endStep]);
 
-  var rangeMin = (start == 0) ? 1 : (start / steps) * max;
-  var rangeMax = (end / steps) * max;
-
-  var inRange
-  if(rangeMin == rangeMax){
-    inRange = 1;
+  var inStepRange;
+  if(startStep == endStep){
+    inStepRange = 0;
```

Comment:
Instead of setting `inStepRange = 0` and doing useless math further below, you just have a: `return pentaho.VizController.getRrbColor(color1[0], color1[1], color1[2]);` here